### PR TITLE
Adapt console selecting to more backends and scenarios

### PR DESCRIPTION
**Pull request by waynechen55**
* **Move** subroutine ```select_backend_console``` from module ```lib/virt_autotest/utils.pm``` to module ```lib/utils.pm```.

* **Adapt** subroutine ```select_backend_console``` to more architectures and backends, including ipmi, qemu, pvm_hmc, svirt, s390x and etc. Please refer to subroutine description for details.

* **Update** subroutine calling in modules:
  * ```lib/guest_installation_and_configuration_base.pm```
  * ```lib/virt_autotest/virtual_network_utils.pm```
  * ```tests/console/system_prepare.pm```
  * ```tests/console/validate_repos.pm```
  * ```tests/virt_autotest/prepare_non_transactional_server.pm```
  * ```tests/virt_autotest/restore_upgrade_system.pm```
  * ```tests/virt_autotest/unified_guest_installation.pm```
  * ```tests/virt_autotest/verify_virtualization.pm```
  * ```tests/yam/migration/zypper_migration.pm```
  * ```tests/yam/validate/validate_connectivity.pm```
  * ```tests/yam/validate/validate_migration_logs.pm```
  * ```tests/yam/validate/validate_product.pm```

* **Please** also review https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/26480 which leads to this pull request if you are interested in host upgrade test.

* **Verification Runs:**
  * [sle-16.1-Online-s390x-Build63.1-sles_default_standard_unattended@s390x-kvm](https://openqa.suse.de/tests/24077505)
  * [sle-16.1-Online-s390x-Build63.1-sles_default_standard_unattended@s390x-zVM](https://openqa.suse.de/tests/24077506)
  * [sle-16.1-Online-s390x-Build63.1-gi-online-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/24077541)
  * [sle-16.1-Online-aarch64-Build63.1-sles_migration_15sp6_textmode](https://openqa.suse.de/tests/24080864)
  * [ sle-16.1-Online-ppc64le-Build63.1-sles_migration_15sp6_textmode_all_modules_and_extensions](https://openqa.suse.de/tests/24080866)
  * [sle-16.1-Online-ppc64le-Build63.1-sles_migration_160](https://openqa.suse.de/tests/24080867)
  * [sle-16.1-Online-s390x-Build63.1-sles_migration_15sp5_textmode](https://openqa.suse.de/tests/24081080)
  * [sle-16.1-Online-x86_64-Build63.1-sles_migration_15sp6_textmode](https://openqa.suse.de/tests/24081083)
  * [sle-16.1-Online-x86_64-Build63.1-sles_sap_migration_15sp6](https://openqa.suse.de/tests/24081084)
  * [sle-16.1-Online-aarch64-Build63.1-sles_default_standard_unattended](https://openqa.suse.de/tests/24081814)
  * [sle-16.1-Online-ppc64le-Build63.1-sles_default_standard_unattended](https://openqa.suse.de/tests/24081818)
  * [sle-16.1-Online-x86_64-Build63.1-sles_default_standard_unattended](https://openqa.suse.de/tests/24081827)
  * [sle-16.1-Online-x86_64-Build63.1-uefi-gi-full-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/24081831)
  * [sle-16.1-Online-aarch64-Build63.1-slmicro_migration](https://openqa.suse.de/tests/24081833)
  * [sle-16.1-Online-aarch64-Build63.1-uefi-gi-guest_postsles16-on-host_developing-kvm](https://openqa.suse.de/tests/24096574)
  * [sle-16.1-Online-aarch64-Build63.1-uefi-gi-guest_developing-on-host_sles15sp7-kvm](https://openqa.suse.de/tests/24096576)
  * [sle-16.1-Online-x86_64-Build63.1-uefi-gi-guest_sles16_0-online-on-host_developing-kvm](https://openqa.suse.de/tests/24102019)

### CODE REVIEW

**Comment by jknphy** [comment]
Please announce this PR in #eng-testing for example to find proper code reviewers, from QE Iam I can give my two cents but I am not expert in console handling in openQA. I tried to remove the rest of the reviewers from our squad (to avoid confusion about who could help here) but they still appear, I think you need to remove them yourself. This have wide scope IMO. I hope it makes sense. Thanks for testing those special backends btw.

**Comment by waynechen55** [comment]
> Please announce this PR in #eng-testing for example to find proper code reviewers, from QE Iam I can give my two cents but I am not expert in console handling in openQA. I tried to remove the rest of the reviewers from our squad (to avoid confusion about who could help here) but they still appear, I think you need to remove them yourself. This have wide scope IMO. I hope it makes sense. Thanks for testing those special backends btw.

I will evaluate and see what I can do.

**Comment by waynechen55** [comment]
Welcome your review @alice-suse @guoxuguang @Julie-CAO @RoyCai7 @nanzhg @tbaev @martinsmarcelo Please help have a look if you are interested and leave a message.

File: lib/utils.pm [hunk 1]
```diff
@@ -3739,4 +3741,92 @@ sub query_installed_packages {
...
+
+    reset_consoles for (0 .. $args{reset_times} - 1);
+    if (is_ipmi) {
+        select_console($args{console}, await_console => $args{wait});
+        use_ssh_serial_console if ($args{init} and $args{ready});
+    }
+    elsif (is_qemu) {
+        console('x11')->set_tty(get_x11_console_tty);
+        console('root-console')->set_tty(get_root_console_tty);
+        reset_consoles;
```
Comment (ldevulder) [inline lib/utils.pm:3833]: Is this `reset_consoles` needed? Has we have one [L3800.](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/26486/changes#diff-47a0a7f15c1bd3166c2ca1c8c3f4f0c7f7beb1b1b5c4d349e4da8cab17e9ae5eR3800)

File: lib/utils.pm [hunk 2]
```diff
@@ -3739,4 +3741,92 @@ sub query_installed_packages {
...
+
+sub select_backend_console {
+    my (%args) = @_;
+    $args{reset_times} //= 1;
+    $args{init} //= 0;
+    $args{wait} //= 1;
+    $args{ready} //= 0;
+
+    if (is_ipmi) {
+        $args{console} //= ($args{init} ? 'sol' : 'root-ssh');
```
Comment (ldevulder) [inline lib/utils.pm]: Maybe this line could be added after [L3801](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/26486/changes#diff-47a0a7f15c1bd3166c2ca1c8c3f4f0c7f7beb1b1b5c4d349e4da8cab17e9ae5eR3801) to avoid having another `if (is_ipmi)`? But maybe I'm wrong and `reset_consoles` needs it?

If we could move this line please consider doing the same for the others like this. But again I may have wrong ;-)

File: lib/utils.pm [hunk 3]
```diff
@@ -3739,4 +3741,92 @@ sub query_installed_packages {
...
+=cut
+
+sub is_textmode {
+    return check_var('DESKTOP', 'textmode') || check_var('TEXTMODE', 1);
+}
+
+=head2 select_backend_console
+
+Select corresponding ipmi, qemu, pvm, svirt or other backend console. If argument
+init is set, select the initial or initialization console to start, for exmaple,
```
Comment (RoyCai7) [inline lib/utils.pm]: s/exmaple/example/

File: lib/utils.pm [hunk 4]
```diff
@@ -3739,4 +3741,92 @@ sub query_installed_packages {
...
+state and etc, especially if system is ready and init is also set, the initial
+and desired consoles will be selected in order (e.g. sol and root-ssh for ipmi).
+User can also specify desired console or choose whether to wait for the selected
+console. 
+C<%args> have following keys to be defined:
+- C<console>: desired console to be selected or use backend default.
+- C<reset_times>: specifies the number of times (default to 1) reset_consoles to 
+                  be done.
+- C<init>: Whether select the initial or initialization console (default to 0).
+           By the way, some backends may not use it at all, for exmaple, qemu
```
Comment (RoyCai7) [inline lib/utils.pm]: s/exmaple/example/

File: lib/utils.pm [hunk 5]
```diff
@@ -3739,4 +3741,92 @@ sub query_installed_packages {
...
+        ensure_serialdev_permissions;
+        serial_terminal::prepare_serial_console();
+    }
+    elsif (is_pvm_hmc) {
+        select_console($args{console}, await_console => $args{wait});
+        select_console('root-ssh') if ($args{init} and $args{ready});
+    }
+    elsif (is_svirt_except_s390x) {
+        select_console($args{console}, await_console => $args{wait});
+        save_svirt_pty if (is_s390x and is_textmode);
```
Comment (RoyCai7) [inline lib/utils.pm]: This line is dead code — it can never execute.

File: lib/utils.pm [hunk 6]
```diff
@@ -3739,4 +3741,92 @@ sub query_installed_packages {
...
+for example, at system bootup, already spilled login prompt, confirm the system
+state and etc, especially if system is ready and init is also set, the initial
+and desired consoles will be selected in order (e.g. sol and root-ssh for ipmi).
+User can also specify desired console or choose whether to wait for the selected
+console. 
+C<%args> have following keys to be defined:
+- C<console>: desired console to be selected or use backend default.
+- C<reset_times>: specifies the number of times (default to 1) reset_consoles to 
+                  be done.
+- C<init>: Whether select the initial or initialization console (default to 0).
```
Comment (alice-suse) [inline lib/utils.pm:3809]: What init parameter serves to specify is unclear to me --  by `the initial or initialization console`, they could be different thing

File: lib/utils.pm [hunk 7]
```diff
@@ -3739,4 +3741,92 @@ sub query_installed_packages {
...
+           By the way, some backends may not use it at all, for exmaple, qemu
+           backend.
+- C<wait>: Whether wait for asserting the selected console (default to 1).
+- C<ready>: Whether system is alread up and running (default to 0).
```
Comment (alice-suse) [inline lib/utils.pm:3813]: The init and ready two parameters seem to affect the final selected console, regardless of whether `console` is given in its call for some backend, eg ipmi, pvm_hmc. This causes confusion of the function.

Not sure if I understand correctly, this function is moved from virt_autotest/utils to higher level lib/utils and extended for more backends -- the purpose is to make it a more general function to serve more backends? If so, suggest to refine the comments to make it clear enough.

Besides, just to double confirm. I noticed you have VRs for different backends. Has this function been really executed in those jobs?

File: tests/console/validate_repos.pm [hunk 8]
```diff
@@ -11,11 +11,12 @@ use Mojo::Base 'consoletest';
 use testapi;
 use repo_tools 'validate_repo_properties';
 use scheduler 'get_test_suite_data';
+use utils 'select_backend_console';
 
 sub run {
     my $test_data = get_test_suite_data();
 
-    select_console 'root-console';
+    select_backend_console;
```
Comment (alice-suse) [inline tests/console/validate_repos.pm:19]: Is it correct understanding that you've replaced all select_console calls with select_backend_console in the scope of host upgrade related files, but not in other scopes? Just asking, no intention to ask for more replacement. 

BTW, IMO, select_console with console name is clearer than this select_backend_console function and will not cause surprise (for those unfamiliar with the func) that the specified console is not selected but other consoles are. **To me, select_backend_console is suitable to replace a code block(eg handles reset_console, select_console of serial, then select ssh in ipmi) but not those single line of select_console. I'd recommend to revert such replacement and keep those where are more suitable to use it.**

Comment (rfan1) [reply tests/console/validate_repos.pm:19]: I will prefer to keep `root_console` as it works fine in past few years. and we are familiar with it.

Comment (waynechen55) [reply tests/console/validate_repos.pm:19]: Different backends  use different consoles, so why ```select_backend_console``` is used. For example, ipmi uses  ```root-ssh``` which is different from ```root-console```(they are totally different). With ```select_backend_console``` different consoles can be selected for different backends and uses which eliminates code block. Without ```select_backend_console```, there will be if else code block. At the same time, you can keep using ```root-console``` without trouble.

Introducing ```select_backend_console``` is not to just replace one single line, it is to avoid similar code block in many different places.

File: lib/utils.pm (hunk 7 above)
Comment (waynechen55) [reply lib/utils.pm:3813]: 1. Yes. I want it to be a more generic function and, at the same time, use a function from virt_autotest lib in yam migration is not that good. Moving it to lib/utils.pm is better.

2. In order to make it a more generic function, I took further consideration about how to make it function more properly. For example, the ```select_backend_console``` itself can select console for different backends and at different phases.  The ```init``` indicates initial or initialization, for example, sol console might be used at that time instead of ssh. The ```ready``` indicates system is already up and running, for example, ssh connection is ready, so ssh console can be selected, but maybe user want to have a look at the sol console first before switching to ssh after coming back from other tasks, so ```init=1``` and ```ready=1``` serve the purpose. Leave ```init``` with default 0, user can just select ssh.

3. Yes, I checked. This function was called in all verifications.

File: lib/utils.pm (hunk 6 above)
Comment (waynechen55) [reply lib/utils.pm:3809]: See comment https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/26486#discussion_r3910311543

File: lib/utils.pm (hunk 5 above)
Comment (waynechen55) [reply lib/utils.pm]: It is not dead code. ```is_svirt_except_s390x``` checks backend vs ```is_s390x``` checks architecture and ```is_textmode``` checks video mode.

File: lib/utils.pm (hunk 7 above)
Comment (waynechen55) [reply lib/utils.pm:3813]: Already add more description for the subroutine.

File: lib/utils.pm (hunk 2 above)
Comment (waynechen55) [reply lib/utils.pm]: You are right. I moved it down and saved one large if...else block.

File: lib/utils.pm (hunk 3 above)
Comment (waynechen55) [reply lib/utils.pm]: Done

File: lib/utils.pm (hunk 4 above)
Comment (waynechen55) [reply lib/utils.pm]: Done

File: tests/console/validate_repos.pm (hunk 8 above)
Comment (rfan1) [reply tests/console/validate_repos.pm:19]: Thanks, then it makes more sense to use the new one!

File: lib/utils.pm (hunk 1 above)
Comment (waynechen55) [reply lib/utils.pm:3833]: Better keep it because this is what is usually done when resetting console tty. I just keep it and I did not invent it.

File: tests/console/validate_repos.pm (hunk 8 above)
Comment (jknphy) [reply tests/console/validate_repos.pm:19]: and why not make the test module agnostics? is that a possibility? Perhaps in a few corner case scenario you need this (switching console in the same module for some reason), but if you have a single module where you switch the console and then the rest following it are do not have any console selection might be a better design. wdyt?

Comment (waynechen55) [reply tests/console/validate_repos.pm:19]: I think a single console selecting subroutine is better and more flexible. It does not depend on test flow at all as well.

Comment (alice-suse) [reply tests/console/validate_repos.pm:19]: > Different backends use different consoles, so why `select_backend_console` is used. For example, ipmi uses `root-ssh` which is different from `root-console`(they are totally different). With `select_backend_console` different consoles can be selected for different backends and uses which eliminates code block. Without `select_backend_console`, there will be if else code block. At the same time, you can keep using `root-console` without trouble.
> 
> Introducing `select_backend_console` is not to just replace one single line, it is to avoid similar code block in many different places.

Okay, now I understand better why it is used for such single console selection -- to avoid if else for different architectures. Make sense.

File: lib/utils.pm (hunk 7 above)
Comment (alice-suse) [reply lib/utils.pm:3813]: Great! Now the length of some comment lines are very long, better to shorten a bit.

Comment (waynechen55) [reply lib/utils.pm:3813]: They are all around 80 characters. No need to shorten.

Comment (alice-suse) [reply lib/utils.pm:3813]: Oh, right. The view yesterday was not correct.

**Review by alice-suse — DISMISSED** [review]

**Comment by waynechen55** [comment]
All additional verification runs passed:

[sle-16.1-Full-aarch64-Build66.1-sles_check_installation_medium](https://openqa.suse.de/tests/24204314)
[sle-16.1-Full-ppc64le-Build66.1-sles_default_standard_unattended](https://openqa.suse.de/tests/24204315)
[sle-16.1-Online-x86_64-Build66.1-sles_tpm_fde_unattended](https://openqa.suse.de/tests/24204321)
[sle-16.1-Online-s390x-Build66.1-sles_migration_15sp5_textmode](https://openqa.suse.de/tests/24204438)
[sle-16.1-Online-ppc64le-Build66.1-sles_migration_15sp7_textmode](https://openqa.suse.de/tests/24204439)
[sle-16.1-Online-ppc64le-Build66.1-sles_migration_15sp7_textmode_all_modules_and_extensions](https://openqa.suse.de/tests/24204440)
[sle-16.1-Online-aarch64-Build66.1-sles_migration_15sp5_textmode_scc](https://openqa.suse.de/tests/24244970)
[ sle-16.1-Online-x86_64-Build66.1-sles_migration_160](https://openqa.suse.de/tests/24204443)
[sle-16.1-Online-s390x-Build66.1-sles_migration_15sp5_textmode](https://openqa.suse.de/tests/24204438)
[sle-16.1-Online-s390x-Build66.1-sles_migration_15sp5_ha](https://openqa.suse.de/tests/24204916)
[sle-16.1-Online-s390x-Build66.1-sles_migration_15sp5_textmode](https://openqa.suse.de/tests/24204917)
[ sle-micro-6.2-Elemental-Iso-aarch64-Build6.2_baremetal-iso-image_8.110-test_iso](https://openqa.suse.de/tests/24205021)
[sle-16.1-Online-aarch64-Build66.1-create_hdd_textmode](https://openqa.suse.de/tests/24206242)
[sle-16.1-Online-aarch64-Build66.1-ovs-server](https://openqa.suse.de/tests/24244973)
[sle-16.1-Online-aarch64-Build66.1-ovs-client](https://openqa.suse.de/tests/24244974)
[sle-16.1-Online-ppc64le-Build66.1-create_hdd_textmode](https://openqa.suse.de/tests/24206246)
[sle-16.1-Online-ppc64le-Build66.1-extra_tests_dracut](https://openqa.suse.de/tests/24206247)
[sle-16.1-Online-s390x-Build66.1-create_hdd_textmode](https://openqa.suse.de/tests/24206248)
[sle-16.1-Online-s390x-Build66.1-extra_tests_webserve](https://openqa.suse.de/tests/24206249)
[sle-16.1-Online-Immutable-x86_64-Build66.1-extra_tests_bootloader](https://openqa.suse.de/tests/24206250)
[sle-16.1-Online-Immutable-s390x-Build66.1-extra_tests_zypper_wrapper](https://openqa.suse.de/tests/24206255)
[sle-16.1-Online-x86_64-Build67.2-default_svirt_sle16@svirt-hyperv2022-uefi](https://openqa.suse.de/tests/24245904)
[sle-16.1-Online-x86_64-Build67.2-default_svirt_sle16@svirt-hyperv2025-uefi](https://openqa.suse.de/tests/24245910)
[sle-16.1-Online-x86_64-Build67.2-default_svirt_sle16@svirt-vmware80](https://openqa.suse.de/tests/24246545)
[sle-16.1-Online-aarch64-Build67.2-sles_migration_15sp7_ha](https://openqa.suse.de/tests/24246520)
[sle-16.1-Online-s390x-Build67.2-sles_migration_15sp7_textmode@waynechen55_os-autoinst-distri-opensuse_wayne_complete_select_backend_console@s390x-zVM](https://openqa.suse.de/tests/24246524)
[sle-16.1-Online-x86_64-Build67.2-sles_migration_160_ha](https://openqa.suse.de/tests/24246525)
[sle-16.1-Online-aarch64-Build67.2-ha_qdevice_supportserver](https://openqa.suse.de/tests/24246535)
[sle-16.1-Online-aarch64-Build67.2-ha_qdevice_node2](https://openqa.suse.de/tests/24246536)
[sle-16.1-Online-aarch64-Build67.2-ha_qdevice_server](https://openqa.suse.de/tests/24246537)
[sle-16.1-Online-aarch64-Build67.2-ha_qdevice_node1](https://openqa.suse.de/tests/24246538)
[sle-16.1-Online-ppc64le-Build67.2-sles16_transactional_installation_virt](https://openqa.suse.de/tests/24248741)
[sle-16.1-Transactional-Base-ppc-4096-ppc64le-Build13.31-sles16_transactional_virtualization](https://openqa.suse.de/tests/24248742)
[sle-16.1-Online-ppc64le-Build67.2-sles_migration_15sp5_textmode](https://openqa.suse.de/tests/24248744)
[sle-16.1-Online-ppc64le-Build67.2-create_hdd_ha_textmode](https://openqa.suse.de/tests/24248745)
[sle-16.1-Transactional-Base-aarch64-Build13.31-sles16_transactional_virtualization](https://openqa.suse.de/tests/24248737)
[sle-16.1-Transactional-Base-qcow-x86_64-Build13.31-sles16_transactional_virtualization](https://openqa.suse.de/tests/24248738)
[sle-16.1-Online-aarch64-Build67.2-virt-host_upgrade-15sp7-to-16.1-ipmi](https://openqa.suse.de/tests/24271093)
[sle-16.1-Online-aarch64-Build67.2-virt-host_upgrade-16.0-to-16.1-ipmi](https://openqa.suse.de/tests/24271094)
[sle-16.1-Online-x86_64-Build67.2-virt-host_upgrade-15sp7-to-16.1-qemu](https://openqa.suse.de/tests/24271096)
[sle-16.1-Online-x86_64-Build67.2-virt-host_upgrade-16.0-to-16.1-qemu](https://openqa.suse.de/tests/24271097)

**Comment by waynechen55** [comment]
Ready for merge @alice-suse @guoxuguang @Julie-CAO @RoyCai7 @nanzhg @tbaev @martinsmarcelo Please help have a look if you are interested and leave a message.

**Review by alice-suse — DISMISSED** [review]

**Review by RoyCai7 — DISMISSED** [review]

**Comment by waynechen55** [comment]
I resolved conflicts and need re-approval. @alice-suse @guoxuguang @Julie-CAO @RoyCai7 @nanzhg @tbaev @martinsmarcelo Please help have a look if you are interested and leave a message.

**Review by Julie-CAO — APPROVED** [review]

**Review by RoyCai7 — APPROVED** [review]

**Comment by DimStar77** [comment]
https://openqa.opensuse.org/tests/6212828#step/system_prepare/9

This kind of error seems related to this PR

**Comment by zhaocongjia** [comment]
https://openqa.suse.de/tests/24326555#step/system_prepare/4

This is another example of breakage.  Not sure about the details yet.

**Comment by Vogtinator** [comment]
Revert PR: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/26651

File: lib/utils.pm [hunk 9]
```diff
@@ -3773,4 +3775,95 @@ sub query_installed_packages {
     return ($ret ? 0 : 1);
 }
 
+=head2 is_textmode
+
+  is_textmode()
+
+Check whether system runs in 'textmode' or not.
```
Comment (Vogtinator) [inline lib/utils.pm:3782]: What's this meant to be used for? Systems with a graphical desktop also have a "textmode" and s390x z/VM has neither.

File: lib/utils.pm [hunk 10]
```diff
@@ -3773,4 +3775,95 @@ sub query_installed_packages {
...
+=cut
+
+sub is_textmode {
+    return check_var('DESKTOP', 'textmode') || check_var('TEXTMODE', 1);
+}
+
+=head2 select_backend_console
+
+This subroutine aims to serve console selecting for different backends, including
```
Comment (Vogtinator) [inline lib/utils.pm:3791]: What is it meant to be used for?

The method name looks like it would select the console of the backend (e.g. ipmi, svirt, z/VM), but that's not what it does.

File: lib/utils.pm [hunk 11]
```diff
@@ -3773,4 +3775,95 @@ sub query_installed_packages {
...
+    $args{ready} //= 0;
+
+    reset_consoles for (0 .. $args{reset_times} - 1);
+    if (is_ipmi) {
+        $args{console} //= ($args{init} ? 'sol' : 'root-ssh');
+        select_console($args{console}, await_console => $args{wait});
+        use_ssh_serial_console if ($args{init} and $args{ready});
+    }
+    elsif (is_qemu) {
+        $args{console} //= (is_textmode ? 'root-console' : 'x11');
```
Comment (Vogtinator) [inline lib/utils.pm:3830]: This can't work: test modules either need X11 or console, I'm not aware of any module that accepts either.

File: lib/utils.pm [hunk 12]
```diff
@@ -3773,4 +3775,95 @@ sub query_installed_packages {
...
+        console('x11')->set_tty(get_x11_console_tty);
```
Comment (Vogtinator) [inline lib/utils.pm:3831]: Some tests change the x11 tty, this will override it and break those tests.

File: lib/utils.pm [hunk 13]
```diff
@@ -3773,4 +3775,95 @@ sub query_installed_packages {
...
+        console('root-console')->set_tty(get_root_console_tty);
+        reset_consoles;
```
Comment (Vogtinator) [inline lib/utils.pm:3833]: This needlessly resets consoles, possibly breaking tests.

**Comment by waynechen55** [comment]
> https://openqa.opensuse.org/tests/6212828#step/system_prepare/9
> 
> This kind of error seems related to this PR

This is because it is not textmode, so it selects x11 console maybe. See https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/26651#issuecomment-5631205580.

**Comment by waynechen55** [comment]
> https://openqa.suse.de/tests/24326555#step/system_prepare/4
> 
> This is another example of breakage. Not sure about the details yet.

This is because it is not textmode, so it selects x11 console maybe. See https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/26651#issuecomment-5631205580.
